rdmo/overlays/viewsets.py

```diff
@@ -10,63 +10,60 @@
 from .models import Overlay
 
 
-class OverlayViewSet(ViewSet):
+def check_overlays_enabled():
+    if not settings.OVERLAYS:
+        raise NotFound("Overlays are disabled")
 
-    @action(detail=False, methods=['post'], url_path='(?P<url_name>[-\\w]+)/current',
-            permission_classes=[IsAuthenticated])
-    def current(self, request, url_name=None):
-        site = Site.objects.get_current()
-        overlays = settings.OVERLAYS.get(url_name)[:]
+def get_overlays_for_url(url_name):
+    overlays = settings.OVERLAYS.get(url_name)
+    if not overlays:
+        raise NotFound()
+    return overlays
 
-        if not overlays:
-            raise NotFound()
+def get_or_create_overlay(user, site, url_name, default_current):
```

Comment:
This could be done using https://docs.djangoproject.com/en/5.0/ref/models/querysets/#get-or-create (I should have done this in the first place).